.release/security-scan.hcl

```
@@ -1,17 +1,33 @@
 # Copyright (c) HashiCorp, Inc.
 # SPDX-License-Identifier: BUSL-1.1
 
+# These scan results are run as part of CRT workflows.
+
+# Un-triaged results will block release. See `security-scanner` docs for more
+# information on how to add `triage` config to unblock releases for specific results.
+# In most cases, we should not need to disable the entire scanner to unblock a release.
+
+# To run manually, install scanner and then from the repository root run
+# `SECURITY_SCANNER_CONFIG_FILE=.release/security-scan.hcl scan ...`
+# To scan a local container, add `local_daemon = true` to the `container` block below.
+# See `security-scanner` docs or run with `--help` for scan target syntax.
+
 container {
 	dependencies = true
 	alpine_secdb = false
```

```suggestion
	alpine_secdb = true
```

To start getting container dependency scanning in place (for `alpine` container image based targets). 

The configuration syntax here is certainly sub-optimal, and may change in the future (with a follow up PR here of course!).

---

Noticing a few fresh vulns flagging here; I expect both will self-resolve with fresh builds in the near future before the next Consul patch release (one isn't even patched yet), but calling out since it looks like this would block a release today. We'll need to add triage blocks if not.

```shell
Scanned docker:{owner:"hashicorp" name:"consul-enterprise"} tag:"local" localDaemon:true in 1m47.3s - found 6 result(s)
  » Dependency Scanner
    ⚠︎ found reported vulnerability CVE-2023-46218 from Alpine Linux's Security Issue Tracker in curl@8.4.0-r0
        /lib/apk/db/installed:0:0
    ⚠︎ found reported vulnerability CVE-2023-46219 from Alpine Linux's Security Issue Tracker in curl@8.4.0-r0
        /lib/apk/db/installed:0:0
    ⚠︎ found reported vulnerability CVE-2023-5678 from Alpine Linux's Security Issue Tracker in openssl@3.1.4-r0
        /lib/apk/db/installed:0:0
    ⚠︎ found reported vulnerability CVE-2023-5678 from Alpine Linux's Security Issue Tracker in openssl@3.1.4-r0
        /lib/apk/db/installed:0:0
    ⚠︎ found reported vulnerability CVE-2023-46218 from Alpine Linux's Security Issue Tracker in curl@8.4.0-r0
        /lib/apk/db/installed:0:0
    ⚠︎ found reported vulnerability CVE-2023-46219 from Alpine Linux's Security Issue Tracker in curl@8.4.0-r0
        /lib/apk/db/installed:0:0
```

@picatz do you have any examples of teams proactively running container scans in prerelease images? I was thinking that might be useful, and this result is making me feel it's maybe worth doing proactively, at least for visibility leading up to releases. We'll get the same feedback when release branches are cut, but that won't leave as much reaction time.

---

FWIW, I know that Boundary has this enabled:
https://github.com/hashicorp/boundary/blob/0adfe2d5a3f4e72e9aaae85922fddb35d47ff343/.release/security-scan.hcl#L4-L6

They've been able to catch these kinds of CVEs quickly because of it in the past. 

> [!TIP]
> We can use the `triage` block in any target block (`binary`, etc) to filter out these results. For example:
> ```hcl
>  container {
>    ...
>    triage {
>        # curl@8.4.0-r0 and openssl@3.1.4-r0 vulnerabilities we can safely ignore
>        vulnerabilities = [ "CVE-2023-46218", "CVE-2023-46219", "CVE-2023-5678" ]
>    }
> }
> ```
> https://github.com/hashicorp/security-scanner/blob/010cdf69a9d8ba1d0b74d1141cbcd49625bd3efe/CONFIG.md?plain=1#L182-L185 
>
> However, please note that any CVE's that no longer show up in the results should be _manually_ removed from the `scan.hcl` contents. This should really only ever be used _temporarily_.